portlist/portlist_linux.go

```diff
@@ -59,25 +88,20 @@ func appendListeningPorts(base []Port) ([]Port, error) {
 		if err != nil {
 			return nil, fmt.Errorf("%s: %s", fname, err)
 		}
-		if br == nil {
-			br = bufio.NewReader(f)
-		} else {
-			br.Reset(f)
-		}
-		ports, err := parsePorts(br, filepath.Base(fname))
+		br.Reset(f)
+		ret, err = appendParsePorts(ret, br, filepath.Base(fname))
 		f.Close()
```

Comment:
I think this results in br being put back in the syncpool still holding a reference to f, so f won't be garbage collected until the next time br gets used and Reset(eofReader).

Which is maybe fine, since it has closed the actual file and so won't hold it open.

Reply:
See the `br.Reset` above:

```go
	br := bufioReaderPool.Get().(*bufio.Reader)
	defer bufioReaderPool.Put(br)
	defer br.Reset(eofReader)
```

That means before it's put back in the pool, we drop the reference to `f`.